src/translator/batch_translator.cpp

```diff
@@ -11,17 +11,30 @@ namespace bergamot {
 BatchTranslator::BatchTranslator(DeviceId const device,
                                  std::vector<Ptr<Vocab const>> &vocabs,
                                  Ptr<Options> options,
-                                 const void * model_memory)
-    : device_(device), options_(options), vocabs_(&vocabs), model_memory_(model_memory) {}
+                                 const void * model_memory,
+                                 const size_t model_memory_size,
+                                 const void * shortlist_memory,
+                                 const size_t shortlist_memory_size)
+    : device_(device), options_(options), vocabs_(&vocabs),
+    model_memory_(model_memory), model_memory_size_(model_memory_size),
+    shortlist_memory_(shortlist_memory), shortlist_memory_size_(shortlist_memory_size) {}
 
 void BatchTranslator::initialize() {
   // Initializes the graph.
   if (options_->hasAndNotEmpty("shortlist")) {
     int srcIdx = 0, trgIdx = 1;
     bool shared_vcb = vocabs_->front() == vocabs_->back();
-    slgen_ = New<data::LexicalShortlistGenerator>(options_, vocabs_->front(),
-                                                  vocabs_->back(), srcIdx,
-                                                  trgIdx, shared_vcb);
+    if (shortlist_memory_ != nullptr && shortlist_memory_size_ > 0) {
+        bool check = true;  // Whether to verify the shortlist content (not sure how to deal with this flag)
```

Comment:
Is this a todo? Are you asking? 

tl;dr: Always check, check logic within marian makes sense.

Service shouldn't be bothered to check the validity of shortlist, there should be a runtime abort from within the shortlist. So if you're getting a struct carrying *data and size, you *should* validate it during shortlist construction.

marian has ABORT for this. If model, shortlist shares similar traits then the validation routines can be moved into marian-dev with some common stuff between them.

If the doubt is regarding whether a `--check-shortlist` argument provided to user - no, always check.

Reply:
Thanks. `check` flag is initially designed when the shortlist is loaded from file. The user (like Mozilla) can decide whether they want to check or not (e.g., for the first time use).  I was struggling with what to do with the binary buffer. I preassume that Mozilla should be very sure that they are passing the right memory to load the shortlist. 

What you said makes sense. I will keep `check` true.

Reply:
check increases shortlist loading time because it has to go over all indices and check they are in range.  I specifically asked for it to be optional.  And honestly it should be.

Reply:
Single `--check-bytearray` (change to name you think is suitable) argument bound to `Service` through `Options`, loaded at configparser - for all, model, shortlist etc. Default to always check. Mozilla has to explicitly disable-for speed, baked into their WASM plumbing.

Below: 
https://github.com/browsermt/bergamot-translator/blob/34228d37bfcc44e36bf01d8ea70fbc2fce9d9ae9/src/translator/parser.h#L22-L25